### DISCUSSION TITLE
Why it is required to use redis >= 7.4 when using Client-side caching

### DISCUSSION THREAD

**Discussion by qingdaoheze** [xetorthio/jedis#3979] [Q&A] [answered]
<img width="1401" alt="image" src="https://github.com/user-attachments/assets/c6f65283-1a83-43e5-939f-97dd6c813135">

https://github.com/redis/jedis/releases/tag/v5.2.0

Because I din't see any update for the Client-side caching in redis 7.4.0

<img width="999" alt="image" src="https://github.com/user-attachments/assets/d9361b80-7733-4775-9da0-6f9c80bb790c">

https://github.com/redis/redis/releases/tag/7.4.0

**Comment by mortensi** [comment, marked as answer]
Redis will soon release support for client-side caching in all its offerings (Redis Software and Redis Cloud), which will be supported only for the 7.4 feature set and later. To avoid inconsistent behavior across Community Edition and the commercial offerings, we decided to streamline the user experience across all the different options. This means developers won't face any incompatibilities while migrating to Redis Software or Redis Cloud. The official announcement will come later.

**Reply by qingdaoheze** [reply]
Thank you for your reply.

**Reply by dzoech** [reply]
Since none of the major cloud providers support Redis 7.4, is there any possibility to utilize client-side caching on those managed instances?

Currently supported are:
- [AWS ElastiCache: 7.1](https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/supported-engine-versions.html)
- [Azure Redis: 7.2](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-upgrade#current-versions)
- [GCP Memorystore: 7.2](https://cloud.google.com/memorystore/docs/redis/supported-versions)

**Reply by dzoech** [reply]
Pinging @mortensi in hope for an answer to my question :)

**Reply by mortensi** [reply]
Hi @dzoech! CSC can be configured where the 7.4 feature set is available (and later versions), so I don't think the feature will work in those managed instances.
